azure/services/managedclusters/spec.go

```diff
@@ -672,3 +680,11 @@ func getIdentity(identity *infrav1.Identity) (managedClusterIdentity *containers
 	}
 	return
 }
+
+// apiServerAccessProfileOrAuthIPRangesIsNil returns true if the managed cluster's APIServerAccessProfile or AuthorizedIPRanges is nil.
+func apiServerAccessProfileOrAuthIPRangesIsNil(managedCluster containerservice.ManagedCluster) bool {
+	if managedCluster.APIServerAccessProfile == nil || managedCluster.APIServerAccessProfile.AuthorizedIPRanges == nil || reflect.DeepEqual(managedCluster.APIServerAccessProfile.AuthorizedIPRanges, &[]string{}) {
+		return true
+	}
+	return false
```

Comment:
Do you think we should add a small unit test for this function so we can test the `return false` case?

Reply:
It could be redundant in adding a test case for this specific function since I am already testing the response of this function in the scenarios I added in spec_test.go. 
What do you think ?

Reply:
Oh, I see. I just pointed it out b/c there was a note from codecov saying it wasn't tested, but if you've already covered it, then we're good.